components/Form.js

```diff
@@ -66,10 +65,13 @@ export default function Form({
 
   const formattedTodayDate = new Date().toISOString().substring(0, 10);
   const [taskToUpdate, setTaskToUpdate] = useState();
-  const [showEndDate, setShowEndDate] = useState(value?.endDate);
+  const [displayEndDate, setDisplayEndDate] = useState(value?.endDate);
+  const [isEndDateValid, setIsEndDateValid] = useState(false);
+  const [endDate, setEndDate] = useState(value?.endDate);
```

Comment:
```suggestion
  const [isEndDateValid, setIsEndDateValid] = useState(false);
  const [endDate, setEndDate] = useState(value?.endDate);
```
I am pretty sure you can use only two states here, as "endDate" and "displayEndDate" maintain the same information :) It seems like 
`  const [displayEndDate, setDisplayEndDate] = useState();` should actually maintain a boolean instead of `value?.endDate`
Please make sure that you need all of these state and what info they should contain.

Either like this, for the conditional rendering:
```suggestion
  const [displayEndDate, setDisplayEndDate] = useState(false);
  const [isEndDateValid, setIsEndDateValid] = useState(false);
  const [endDate, setEndDate] = useState(value?.endDate);
```
Or maybe even
```suggestion
  const [displayEndDate, setDisplayEndDate] = useState(false);
  const [endDate, setEndDate] = useState(value?.endDate);
```
If the states displayEndDate and isEndDateValid are always the same, they can be one state also :)